org.scala-ide.sdt.core.tests/src/org/scalaide/core/semantichighlighting/classifier/RegionParser.scala

```diff
@@ -2,10 +2,44 @@ package org.scalaide.core.semantichighlighting.classifier
 
 import org.eclipse.jface.text.IRegion
 import org.eclipse.jface.text.Region
+import scala.annotation.tailrec
 
 object RegionParser {
 
   /**
+   * This class represents a substring with an optional prefix and suffix.
+   */
+  case class EmbeddedSubstr(str: String, prefix: String = "", suffix: String = "") {
+    private[RegionParser] val searchString = prefix + str + suffix
+  }
+
+  object EmbeddedSubstr {
+    implicit def wrapAsEmbeddedSubstring(str: String) = EmbeddedSubstr(str)
```

Comment:
This already works because the implicit conversion is defined in the companion object; try it yourself:

```
scala> import org.scalaide.core.semantichighlighting.classifier.RegionParser._
scala> val x: EmbeddedSubstr = "3"
res0: EmbeddedSubstr(3,,)
```